migration-engine/migration-engine-tests/tests/migrations/id.rs

```diff
@@ -184,7 +185,8 @@ fn models_with_an_autoincrement_field_as_part_of_a_multi_field_id_can_be_created
 }
 
 // Ignoring sqlite is OK, because sqlite integer primary keys are always auto-incrementing.
-#[test_connector(exclude(Sqlite, CockroachDb))]
+// Ignoring TiDB, we can't set auto_increment for an existing id field to auto-incrementing in TiDB.
+#[test_connector(exclude(Sqlite, CockroachDb, TiDB))]
```

Comment:
Is there a workaround for this? Or an alternative strategy?

Reply:
@Mini256  I think you skipped this ?

Reply:
Related issue: https://github.com/pingcap/tidb/issues/39143

Reply:
From the existing test code, we know that:

> MySQL cannot add autoincrement property to a column that already has data.

MySQL allows this operation when the table has no data, but for the TiDB does not allow setting the `auto_increment` attribute for an existing primary key column even when there is no data for now.

Because it is not easy to determine whether the condition "a table has no data" is satisfied in a distributed environment, we need a mechanism to ensure consistency. So this behavior of MySQL may not be compatible in the near future.

Reply:
Interesting. Thanks!